lib/queue_classic_plus/base.rb

```diff
@@ -165,5 +189,26 @@ def self.deserialized(args)
     def self.execute(sql, *args)
       QC.default_conn_adapter.execute(sql, *args)
     end
+
+    def self.define_callback(wrapped_method, type, callback_method, &_block)
+      callback_module = Module.new
+
+      callback_module.send(:define_method, wrapped_method) do |*args|
+        callback_args = args
+        if wrapped_method == :enqueue
+          callback_args = callback_args.drop(1) # Class/method is the first argument. Callbacks don't need that.
+        end
+
+        if [:before, :around].include?(type)
+          block_given? ? yield(*callback_args) : send(callback_method, *callback_args)
+        end
+        super(*args)
+        if [:after, :around].include?(type)
```

Comment:
I think the behaviour of the `around` is surprising and therefore wrong. I'd expect an `around` block to yield / call the thing it's wrapping, See https://api.rubyonrails.org/classes/ActiveJob/Callbacks/ClassMethods.html#method-i-around_enqueue

The problem with this approach is the `around` callback doesn't know if it's being called before or after the  method - it also can't share context?

I'd suggest deleting the `around` implementation because we don't need it and so changing the behaviour isn't worth it?

Reply:
Yep, totally misinterpreted the semantics of `around`. Removed.